### PR TITLE
PUBDEV-7573 DeepLearning cross validation fix for distribution AUTO representing Huber

### PR DESCRIPTION
**Pull request by koniecsveta**
https://0xdata.atlassian.net/browse/PUBDEV-7573

### CODE REVIEW

File: h2o-core/src/main/java/hex/Model.java [hunk 1]
```diff
@@ -2876,4 +2876,11 @@ protected boolean isFeatureUsedInPredict(int featureIdx) {
             .filter(this::isFeatureUsedInPredict)
             .toArray(String[]::new);
   }
+  
+  public boolean isDistributionHuber() {
+    if (_parms._distribution == DistributionFamily.huber) {
```
Comment (honzasterba) [inline h2o-core/src/main/java/hex/Model.java:2881]: return _parms._distribution == DistributionFamily.huber

File: h2o-core/src/main/java/hex/ModelBuilder.java [hunk 2]
```diff
@@ -843,6 +843,13 @@ public void cv_mainModelScores(int N, ModelMetrics.MetricBuilder mbs[], ModelBui
     // Now, the main model is complete (has cv metrics)
     DKV.put(mainModel);
   }
+  
+  public boolean isDistributionHuber() {
```
Comment (honzasterba) [inline h2o-core/src/main/java/hex/ModelBuilder.java:847]: can we just keep the one on modelbuilder or model? do we need both?

File: h2o-algos/src/main/java/hex/deeplearning/DeepLearning.java [hunk 3]
```diff
@@ -602,4 +603,13 @@ else if (tspi == -1) {
     assert(tspi != 0 && tspi != -1 && tspi != -2 && tspi >= 1);
     return tspi;
   }
+  
+  @Override
+  public boolean isDistributionHuber() {
+    if (_parms._distribution == DistributionFamily.huber
```
Comment (honzasterba) [inline h2o-algos/src/main/java/hex/deeplearning/DeepLearning.java:609]: also could bbe just return without if

File: h2o-core/src/main/java/hex/ModelBuilder.java (hunk 2 above)
Comment (koniecsveta) [reply h2o-core/src/main/java/hex/ModelBuilder.java]: hm yesyes we can have it just on model

File: h2o-algos/src/main/java/hex/deeplearning/DeepLearningModel.java [hunk 4]
```diff
@@ -2305,5 +2305,10 @@ public boolean isFeatureUsedInPredict(String featureName) {
     int featureIdx = ArrayUtils.find(varImp()._names, featureName);
     return featureIdx != -1 && (double) varImp()._varimp[featureIdx] != 0d;
   }
+
+  @Override
+  public boolean isDistributionHuber() {
+    return _parms._distribution == DistributionFamily.huber || get_params()._distribution == DistributionFamily.huber;
```
Comment (michalkurka) [inline h2o-algos/src/main/java/hex/deeplearning/DeepLearningModel.java:2311]: I think this should call super.isDistributionHuber, eg. super.isDistributionHuber || get_params()._ ...

File: h2o-algos/src/test/java/hex/deeplearning/DeepLearningTest.java [hunk 5]
```diff
@@ -1956,10 +1992,26 @@ public void testHuber() {
...
+
+      dl2 = new DeepLearning(parms2).trainModel().get();
+
+      Assert.assertEquals(6.4964976811,((ModelMetricsRegression)dl2._output._training_metrics)._mean_residual_deviance,1e-5);
+
     } finally {
       if (tfr != null) tfr.delete();
       if (dl != null) dl.deleteCrossValidationModels();
       if (dl != null) dl.delete();
+        if (dl2 != null) dl2.deleteCrossValidationModels();
```
Comment (honzasterba) [inline h2o-algos/src/test/java/hex/deeplearning/DeepLearningTest.java:2013]: indent

**Pull request merged**